redash/serializers/query_result.py

```diff
@@ -57,6 +57,14 @@ def _get_column_lists(columns):
     return fieldnames, special_columns
 
 
+def serialize_query_result(query_result, is_api_user):
+    if is_api_user:
+        publicly_needed_keys = ['data', 'retrieved_at']
+        return {key: query_result.to_dict()[key] for key in publicly_needed_keys}

```

Comment:
Current implementation will call `to_dict` multiple times, which will result in calling `json_loads` multiple times.

You should make sure to call `to_dict` only once, and you can also use [`project`](https://funcy.readthedocs.io/en/stable/colls.html?highlight=project#project) instead of "manual" extraction.

Reply:
You've shown me `project` before, and I was looking for it with the wrong search term (looked for dict masking in Python, but all examples were soooo complicated). Thanks!